src/hera/v1/operator.py

```diff
@@ -0,0 +1,33 @@
+class Operator:
```

Comment:
This is awesome! I do have a suggestion, though. I think we can simplify this to an enum, like so:
```python
class Operator(str, Enum):
    equals = '=='
    greater = '>'
    lower = '<'
    greater_equal = '>='
    lower_equal = '<='
    not_equal = '!='
``` 

Let me know what you think!

Reply:
I've tried that and got TypeError, because `when`function is expecting `Operator` and if we do `.when(t1, Operator.equals, "t2"` we give an `str` input instead.

Reply:
I didn't used `Enum` in this case.. i'll try that!

Reply:
I've updated the implementation, if we use `Operator(str, Enum):` then:
```bash
$ make verify
flake8 src
mypy --namespace-packages --explicit-package-bases -p hera
src/hera/v1/operator.py:17: error: Incompatible types in assignment (expression has type "str", base class "str" defined the type as "Callable[[str], str]")
Found 1 error in 1 file (checked 17 source files)
```
just `Enum` works fine..